modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartClientCodegen.java

```diff
@@ -202,7 +202,10 @@ public void processOpts() {
         } else {
             // dart 2.x
             LOGGER.info("Dart version: 2.x");
-            embeddedTemplateDir = templateDir = "dart2";
+            // check to not overwrite a custom templateDir
+            if (templateDir == null) {
+                embeddedTemplateDir = templateDir = "dart2";
+            }
```

Comment:
I think this may be causing issues. I ran `./bin/dart2-petstore.sh` and got the following diff (partial):
```
--- a/samples/client/petstore/dart2/flutter_petstore/openapi/.travis.yml
+++ b/samples/client/petstore/dart2/flutter_petstore/openapi/.travis.yml
@@ -3,7 +3,7 @@
 language: dart
 dart:
 # Install a specific stable release
-- "2.2.0"
+- "1.24.3"
 install:
 - pub get
 
diff --git a/samples/client/petstore/dart2/flutter_petstore/openapi/README.md b/samples/client/petstore/dart2/flutter_petstore/openapi/README.md
index e78e0e3e69..8520a219f8 100644
--- a/samples/client/petstore/dart2/flutter_petstore/openapi/README.md
+++ b/samples/client/petstore/dart2/flutter_petstore/openapi/README.md
@@ -44,10 +44,10 @@ Please follow the [installation procedure](#installation--usage) and then run th
 import 'package:openapi/api.dart';
 
 // TODO Configure OAuth2 access token for authorization: petstore_auth
-//defaultApiClient.getAuthentication<OAuth>('petstore_auth').accessToken = 'YOUR_ACCESS_TOKEN';
+//openapi.api.Configuration.accessToken = 'YOUR_ACCESS_TOKEN';
 
-var api_instance = PetApi();
-var body = Pet(); // Pet | Pet object that needs to be added to the store
+var api_instance = new PetApi();
+var body = new Pet(); // Pet | Pet object that needs to be added to the store
 
 try {

```

so dart 2 seems to be using dart 1.x template instead.

Reply:
I would suggest you remove these changes so that we can merge the remaining changes into master to give us more time reviewing this particular issue.

Reply:
I think I know what the problem is and will file another PR to fix it after merging your PR